ament_mypy/ament_mypy/main.py

```diff
@@ -0,0 +1,257 @@
+    )
+
+    if error_messages:
+        print('mypy error encountered', file=sys.stderr)
+        print(error_messages, file=sys.stderr)
+        return exit_code
+
+    errors_parsed = get_errors(normal_report)
+
+    print('\n{} files checked'.format(len(filenames)))
+    if not normal_report:
+        print('No errors found')
+    else:
+        print('{} errors'.format(len(errors_parsed)))
+
+    print(normal_report)
+
+    print('\nChecked files:')
+    print(''.join(['\n* {}'.format(f) for f in filenames]))
+
+    # generate xunit file
+    if args.xunit_file:
+        folder_name = os.path.basename(os.path.dirname(args.xunit_file))
+        file_name = os.path.basename(args.xunit_file)
+        suffix = '.xml'
+        if file_name.endswith(suffix):
+            file_name = file_name[:-len(suffix)]
+            suffix = '.xunit'
+            if file_name.endswith(suffix):
+                file_name = file_name[:-len(suffix)]
+        testname = '{}.{}'.format(folder_name, file_name)
+
+        xml = get_xunit_content(errors_parsed, testname, filenames, time.time() - start_time)
+        path = os.path.dirname(os.path.abspath(args.xunit_file))
+        if not os.path.exists(path):
+            os.makedirs(path)
+        with open(args.xunit_file, 'w') as f:
+            f.write(xml)
+
+    return exit_code
+
+
+def generate_mypy_report(config_file: str,
+                         paths: List[str],
+                         cache_dir: str) -> Tuple[str, str, int]:
+    mypy_argv = []
+    mypy_argv.append('--cache-dir')
+    mypy_argv.append(str(cache_dir))
+    if cache_dir == os.devnull:
+        mypy_argv.append('--no-incremental')
+    if config_file is not None:
+        mypy_argv.append('--config-file')
+        mypy_argv.append(str(config_file))
+    if not os.environ.get('MYPYPATH') and os.environ.get('PYTHONPATH') and os.environ.get('COLCON_PREFIX_PATH'):
+        # Filter PYTHONPATHs by colcon prefixes
+        python_paths = os.environ['PYTHONPATH'].split(':')
+        colcon_paths = os.environ['COLCON_PREFIX_PATH'].split(':')
+        intersecting_paths = []
+        for python_path in python_paths:
+            for colcon_path in colcon_paths:
+                # The colcon prefix is pointing at the install space, but ideally we could extract
+                # both install-space and build-space paths from PYTHONPATH. There doesn't seem to be
+                # a way to extract these programatically, so we'll assume that the defaults are being
+                # used and the install- and build-space share a parent directory. This assumption will
+                # not be valid if the `--build-base` or `--install-base` options were used for
+                # `colcon build`.
+                if python_path and python_path.startswith(os.path.dirname(colcon_path)):
+                    intersecting_paths.append(python_path)
+                    break
+
+        os.environ['MYPYPATH'] = ':'.join(intersecting_paths)
```

Comment:
Should the separator be platform specific? On Windows `;`?

Reply:
Ah, looks like it needs to be [`os.pathsep`](https://github.com/python/mypy/blob/404d5c97ce6f0f32acc687ee5695371eac0cba94/mypy/modulefinder.py#L353).

Reply:
fixed in 538014f